SpatialGDK/Source/SpatialGDKTests/Examples/SpatialGDKTestGuidelines.h

```diff
@@ -0,0 +1,69 @@
+
+2. Folder structure
+	- Tests are located in a separate SpatialGDKTests module.
+	So for every component that needs to be tested - it has to be exposed via corresponding macro.
+	E.g. `class FRPCContainer` -> `class SPATIALGDK_API FRPCContainer`, to make FRPCContainer testable.
+	WThe macro is different in each module SPATIALGDK_API, SPATIALGDKSERVICES_API etc.
+
+	- The folder structure inside SpatialGDKTests should resemble the folder structure for components being tested.
+	E.g. If file tested is
+	`UnrealGDK\SpatialGDK\Source\SpatialGDK\Public\Utils\RPCContainer.cpp`,
+	then the corresponding test should be located in
+	`UnrealGDK\SpatialGDK\Source\SpatialGDKTests\SpatialGDK\Utils\RPCContainer\RPCContainerTest.cpp`.
+	There should be a folder with the same name as component tested. It should include the test and all the supporting files.
+
+	- In case of integration tests, that involve multiple components - they should be located in
+	`UnrealGDK\SpatialGDK\Source\SpatialGDKTests\IntegrationTests\<Custom Test Folder>` folder.
+
+3. Test definitions (check TestDefinitions.h for more info)
+    - We have defined 3 types of Macro to be used when writing tests:
+	(https://docs.unrealengine.com/en-US/Programming/Automation/TechnicalGuide/index.html has more information)
+	TEST - a simple test, that should be used if it's one of a kind (i.e. it's body can't be reused, otherwise use COMPLEX_TEST),
+	and if doesn't rely on background threads doing the computation (otherwise use LATENT_COMMANDs).
+	COMPLEX_TEST - same as simple test, but allows having multiple test cases run through the same test function body.
+	DEFINE_LATENT_COMMAND... - used to run tests that are expected to run across multiple ticks.
+	Latent command names should start with `F` (e.g. DEFINE_LATENT_AUTOMATION_COMMAND(FStartDeployment)").
+
+	- There are 5 types of mock objects we can use in tests:
+	Dummy objects
+		are passed around but never actually used. Usually they are just used to fill parameter lists.
+
+	Fake objects
+		actually have working implementations, but usually take some shortcut which makes them not suitable for production (an InMemoryTestDatabase is a good example).
+
+	Stubs
+		provide canned answers to calls made during the test, usually not responding at all to anything outside what's programmed in for the test.
+
+	Spies
+		are stubs that also record some information based on how they were called.
+		One form of this might be an email service that records how many messages it was sent.
+
+	Mocks
+		are pre-programmed with expectations which form a specification of the calls they are expected to receive.
+		They can throw an exception if they receive a call they don't expect and are checked during verification to ensure they got all the calls they were expecting.
+
+4. Test naming convention
+	- `GIVEN_WHEN_THEN` should be used.
+	E.g. `GIVEN_one_and_two_WHEN_summed_THEN_the_sum_is_three`
+	The number TestTrue/TestFalse in Tests should be kept minimal.
```

Comment:
This line should probably be moved under a new category, and I'd rephrase it like this:

5. Test coverage
- Unit tests should be isolated: Tests should be runnable on any machine, in any order, without affecting each other. If possible, tests should have no dependencies on environmental factors or global/external state. 
- Unit tests should verify a single use-case or code path: they are simpler and more understandable, and that is good for maintainability and debugging. 
- Unit tests should use a minimal (1, when possible) number of TestTrue/TestFalse assertions, that covers only what is needed for the use-case/code path you are testing.

Sources:
- https://dzone.com/articles/unit-testing-best-practices-how-to-get-the-most-ou
- https://www.martinfowler.com/bliki/UnitTest.html

Reply:
Added.